src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,137 @@
+            return GetResourceLockAsync(resourceLock.Id, resourceLock.Resource.EntityId, resourceLock.Resource.EntityType, cancellationToken);
+        }
+        public Task<ResourceLock> GetResourceLockAsync(string lockId, string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) => GetResourceLockAsync(lockId, resourceId, resourceType.ToString(), cancellationToken);
+        public async Task<ResourceLock> GetResourceLockAsync(string lockId, string resourceId, string resourceType, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return await GetAsync<ResourceLock>(lockId, queryParameters.ToString(), nameof(GetResourceLockAsync), lockId, cancellationToken);
+        }
+
+        public Task<string> GetResourceLockRawAsync(string resourceId, string lockId, string resourceType, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return GetRawAsync(lockId, queryParameters.ToString(), nameof(GetResourceLockRawAsync), lockId, cancellationToken);
+        }
+
+        public Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) => GetResourceLocksAsync(resourceId, resourceType.ToString(), cancellationToken);
+        public async Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, string resourceType,  CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return await GetAsync<List<ResourceLock>>(null, queryParameters.ToString(), nameof(GetResourceLocksAsync), null, cancellationToken);
+        }
+
+        public Task<string> GetResourceLocksRawAsync(string resourceId, string resourceType, CancellationToken cancellationToken = default)
```

Comment:
Change this to just having a `string queryString` parameter.

Reply:
why? this will get modified when EM fixes it, and for now, these are the parameters.

Reply:
Personally I'd prefer the raw methods to only have raw inputs, a string content and a string queryString if applicable. I don't like that we're adding extra sugar for the raw methods and by not including a queryString parameter users cannot expand on the query parameters a user sends.

Reply:
oops...this is the raw...missed that.